bee-network/src/service/service.rs

```diff
@@ -0,0 +1,486 @@
+            peerlist.update_info(&peer_id, |info| info.relation = to)?;
+        }
+
+        Command::DialAddress { address } => {
+            let _ = senders.internal_commands.send(Command::DialAddress { address });
+        }
+
+        Command::DialPeer { peer_id } => {
+            let _ = senders.internal_commands.send(Command::DialPeer { peer_id });
+        }
+
+        Command::DisconnectPeer { peer_id } => {
+            disconnect_peer(peer_id, senders, peerlist).await?;
+        }
+
+        Command::RemovePeer { peer_id } => {
+            remove_peer(peer_id, senders, peerlist).await?;
+        }
+
+        Command::UnbanAddress { address } => {
+            let mut peerlist = peerlist.0.write().await;
+            peerlist.unban_address(&address)?;
+
+            let _ = senders.events.send(Event::AddressUnbanned { address });
+        }
+
+        Command::UnbanPeer { peer_id } => {
+            let mut peerlist = peerlist.0.write().await;
+            peerlist.unban_peer(&peer_id)?;
+
+            let _ = senders.events.send(Event::PeerUnbanned { peer_id });
+        }
+    }
+
+    Ok(())
+}
+
+async fn process_internal_event(
+    int_event: InternalEvent,
+    senders: &Senders,
+    peerlist: &PeerList,
+) -> Result<(), PeerError> {
+    match int_event {
+        InternalEvent::AddressBound { address } => {
+            let _ = senders.events.send(Event::AddressBound { address });
+        }
+
+        InternalEvent::ProtocolDropped { peer_id } => {
+            let mut peerlist = peerlist.0.write().await;
+
+            // Try to disconnect, but ignore errors in-case the peer was disconnected already.
+            let _ = peerlist.update_state(&peer_id, |state| state.set_disconnected());
+
+            // Try to remove unknown peers.
+            let _ = peerlist.remove_if(&peer_id, |peer_info, _| peer_info.relation.is_unknown());
+
+            let _ = senders.events.send(Event::PeerDisconnected { peer_id });
+        }
+
+        InternalEvent::ProtocolEstablished {
+            peer_id,
+            peer_addr,
+            conn_info,
+            gossip_in,
+            gossip_out,
+        } => {
+            let mut peerlist = peerlist.0.write().await;
+
+            // In case the peer doesn't exist yet, we create a `PeerInfo` for that peer on-the-fly.
+            if !peerlist.contains(&peer_id) {
+                let peer_info = PeerInfo {
+                    address: peer_addr,
+                    alias: alias!(peer_id).to_string(),
+                    relation: PeerRelation::Unknown,
+                };
+
+                peerlist
+                    .insert_peer(peer_id, peer_info.clone())
+                    .map_err(|(_, _, e)| e)?;
+
+                let _ = senders.events.send(Event::PeerAdded {
+                    peer_id,
+                    info: peer_info,
+                });
+            }
+
+            assert!(peerlist.contains(&peer_id));
```

Comment:
Worth documenting what invariant is being checked here and why so that future maintainers know why this should pass?

Reply:
old code. not necessary anymore. However, I documented the `unwrap` below by explaining that at this point the invariant "peer_id exists in peer_list" is upheld.